app/scripts/lib/multichain-api/wallet-getPermissions.test.js

```diff
@@ -0,0 +1,282 @@
+import { CaveatTypes } from '../../../../shared/constants/permissions';
+import {
+  Caip25CaveatType,
+  Caip25EndowmentPermissionName,
+} from './caip25permissions';
+import { getPermissionsHandler } from './wallet-getPermissions';
+
+const baseRequest = {
+  origin: 'http://test.com',
+};
+
+const createMockedHandler = () => {
+  const next = jest.fn();
+  const end = jest.fn();
+  const getPermissionsForOrigin = jest.fn().mockReturnValue(
+    Object.freeze({
+      eth_accounts: {
+        id: '1',
+        parentCapability: 'eth_accounts',
+        caveats: [
+          {
+            value: ['0xdead', '0xbeef'],
+          },
+        ],
+      },
+      [Caip25EndowmentPermissionName]: {
+        id: '2',
+        parentCapability: Caip25EndowmentPermissionName,
+        caveats: [
+          {
+            type: Caip25CaveatType,
+            value: {
+              requiredScopes: {
+                'eip155:1': {
+                  methods: [],
+                  notifications: [],
+                  accounts: ['eip155:1:0x1', 'eip155:1:0x2'],
+                },
+                'eip155:5': {
+                  methods: [],
+                  notifications: [],
+                  accounts: ['eip155:5:0x1', 'eip155:5:0x3'],
+                },
+              },
+              optionalScopes: {
+                'eip155:1': {
+                  methods: [],
+                  notifications: [],
+                  accounts: ['eip155:1:0xdeadbeef'],
+                },
+              },
+            },
+          },
+        ],
+      },
+      otherPermission: {
+        id: '3',
+        parentCapability: 'otherPermission',
+        caveats: [
+          {
+            value: {
+              foo: 'bar',
+            },
+          },
+        ],
+      },
+    }),
+  );
+  const response = {};
+  const handler = (request) =>
+    getPermissionsHandler.implementation(request, response, next, end, {
+      getPermissionsForOrigin,
+    });
+
+  return {
+    response,
+    next,
+    end,
+    getPermissionsForOrigin,
+    handler,
+  };
+};
+
+describe('getPermissionsHandler', () => {
+  beforeAll(() => {
+    delete process.env.BARAD_DUR;
+  });
```

Comment:
nit: should this be in its own BARAD_DUR flag is not set describe block?

Reply:
done here https://github.com/MetaMask/metamask-extension/pull/26054/commits/65e75c1fa84935320c609e5017e4727e9e99c340